client/cli/go/cmds/heketi_storage.go

```diff
@@ -152,6 +159,12 @@ func createHeketiStorageVolume(c *client.Client, dt api.DurabilityType, replicaC
 			req.GlusterVolumeOptions, "user.heketi.dbstoragelevel custom")
 	}
 
+	if appendStorageOptions == "" {
+		req.GlusterVolumeOptions = append(
+			req.GlusterVolumeOptions,
+			strings.Split(appendStorageOptions, ",")...)
```

Comment:
I think we need to set the storage level to custom even in append mode.

Reply:
What I was trying to do is provide the user with two different semantics, the first and pre existing option is "here are my volume options" this clearly calls for a "custom" level because the user is now overriding the system's default values. The new option I was trying to give the caller a way to say "on top of the defaults do X". Obviously there's a bit of confusion if you supply one of the options that the new defaults include (with a different value). I had hoped to keep the code simple with the idea that even if the user had done that they were accepting that in the future if they didn't override the default options they were opting in to having those "defaults" changed.
A fancier way to do this would be to look thru both lists and if an option in the append list appeared in the defaults list it would warn and or set the options to custom but I'm still not sure I like that semantically and its certainly way more complicated.
I do not think I want to change the semantics of this new option. So I think we either need to drop it from this PR entirely or leave it as is. Thoughts?

Reply:
If we keep the option as-is I can add a comment to the code explaining the semantics as I did here.

Reply:
IMO, any user input should be considered as custom storage level. 

Of the three options
a. heketi defaults
b. user options 
c. user options appended to defaults 

I would prefer to drop "b. user options" as (c) can be used to override the defaults if required.
In that approach, I had https://github.com/raghavendra-talur/heketi/commit/0fe666d0edcbd9b45e47938231c135c472d741b1#diff-7c36e16fac4d1fd7b56c539699be42dfR151 where in the append case all default options are taken but the last option which is storage level.

If you think that is confusing, we can drop the append option.

Reply:
I disagree and I think it might come down how we see the "level" setting. I don't see the "custom" level as saying the user set some customizations. It's "not safe for heketi to automatically change". Obviously, if the user uses the append option they're applying customizations but they are customizations that they're OK with the system changing in the future (if needed). I see stuff like the arbiter option belonging there versus these perf xlator changes that are part of the new defaults.

Reply:
Ok, that clears the confusion about what "custom" means. I think we can leave the code as it is in this PR then.